src/utils/diff/parsers/common.ts

```diff
@@ -149,17 +184,30 @@ export abstract class BaseParser {
         return { added, removed }
     }
 
-    match(content: string): RegExpExecArray | null {
-        return this.buildRegexPattern().exec(content)
+    match(content: string): MatchResult | null {
+        const pattern = this.buildRegexPattern()
+
+        const matches = pattern.exec(content)
+        if (matches) {
+            const varName = matches[1] || matches[2]
+            return {
+                isUnknown: !varName.match(/^["'].*["']$/),
+                // position or named parameter match
+                name: varName.replace(/^["']|["']$/g, ''),
+                matches: matches
```

Comment:
we could just add the content and index directly to this object, rather than the entire RegExpExecArray